src/dxtbx/nexus/__init__.py

```diff
@@ -107,16 +107,14 @@ def get_dxtbx_scan(
 
     oscillation = (0, 0)
     if is_rotation:
+        start = scan_axis[0].to("degree").magnitude
         if scan_axis.end:
-            end = scan_axis.end[0]
+            step = np.median(scan_axis.end[()] - scan_axis[()]).to("degree").magnitude
         elif num_images > 1:
-            end = scan_axis[1]
+            step = np.median(np.diff(scan_axis)).to("degree").magnitude
```

Comment:
`np.diff` appears to downcast to an `ndarray`, as demonstrated by the following warning and traceback:

```
$ dials.show $(dials.data get -q vmxi_thaumatin)/image_15799_master.h5
DIALS (2018) Acta Cryst. D74, 85-97. https://doi.org/10.1107/S2059798317017235
/dls/science/users/hko55533/software/cctbx/conda_base/lib/python3.9/site-packages/numpy/core/_asarray.py:171: UnitStrippedWarning: The unit of the quantity is stripped when downcasting to ndarray.
  return array(a, dtype, copy=False, order=order, subok=True)
Traceback (most recent call last):
  File "/dls/science/users/hko55533/software/cctbx/build/../modules/dials/src/dials/command_line/show.py", line 695, in <module>
    run()
  File "/dls/science/users/hko55533/software/cctbx/conda_base/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/dls/science/users/hko55533/software/cctbx/build/../modules/dials/src/dials/command_line/show.py", line 208, in run
    params, options = parser.parse_args(args=args, show_diff_phil=True)
  File "/dls/science/users/hko55533/software/cctbx/modules/dials/src/dials/util/options.py", line 867, in parse_args
    params, args = self._phil_parser.parse_args(
  File "/dls/science/users/hko55533/software/cctbx/modules/dials/src/dials/util/options.py", line 550, in parse_args
    importer = Importer(
  File "/dls/science/users/hko55533/software/cctbx/modules/dials/src/dials/util/options.py", line 202, in __init__
    self.unhandled = self.try_read_experiments_from_images(
  File "/dls/science/users/hko55533/software/cctbx/modules/dials/src/dials/util/options.py", line 279, in try_read_experiments_from_images
    experiments = ExperimentListFactory.from_filenames(
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/model/experiment_list.py", line 551, in from_filenames
    imageset = format_class.get_imageset(
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/FormatMultiImage.py", line 161, in get_imageset
    format_instance = cls.get_instance(filenames[0], **format_kwargs)
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/Format.py", line 274, in get_instance
    Class._current_instance_ = Class(filename, **kwargs)
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/FormatNXmxDLS.py", line 65, in __init__
    super().__init__(image_file, **kwargs)
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/FormatNXmx.py", line 30, in __init__
    super().__init__(image_file, **kwargs)
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/FormatHDF5.py", line 16, in __init__
    Format.__init__(self, image_file, **kwargs)
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/Format.py", line 168, in __init__
    self.setup()
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/Format.py", line 178, in setup
    self._start()
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/FormatNXmxDLS.py", line 68, in _start
    super()._start()
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/format/FormatNXmx.py", line 46, in _start
    self._scan_model = dxtbx.nexus.get_dxtbx_scan(nxsample, nxdetector)
  File "/dls/science/users/hko55533/software/cctbx/modules/dxtbx/src/dxtbx/nexus/__init__.py", line 114, in get_dxtbx_scan
    step = np.diff(scan_axis).median().to("degree").magnitude
AttributeError: Please report this error to dials-support@lists.sourceforge.net: 'numpy.ndarray' object has no attribute 'median'
```
Instead, "manually" calculating the differences appears to retain the units:
```suggestion
            step = np.median(scan_axis[1:] - scan_axis[:-1]).to("degree").magnitude
```

Reply:
It's a consequence of [`h5py.Dataset` having an `__array__` method](https://github.com/h5py/h5py/blob/c98ba1165f9eb880dd588a2c8a6be180e6bae691/h5py/_hl/dataset.py#L1034-L1047), which we don't overload in the definition of `NXtransformationsAxis`.